File: ios/App/App/AppGroupStoragePlugin.m
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(AppGroupStoragePlugin, "AppGroupStorage",
    CAP_PLUGIN_METHOD(saveToken, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(getToken, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(removeToken, CAPPluginReturnPromise);
)

File: ios/App/App/AppGroupStoragePlugin.swift
import Foundation
import Capacitor

@objc(AppGroupStoragePlugin)
public class AppGroupStoragePlugin: CAPPlugin {
    @objc func saveToken(_ call: CAPPluginCall) {
        guard let token = call.getString("token") else {
            call.reject("Token is required")
            return
        }

        AppGroupStorage.shared.saveToken(token)
        call.resolve()
    }

    @objc func getToken(_ call: CAPPluginCall) {
        if let token = AppGroupStorage.shared.getToken() {
            call.resolve(["token": token])
        } else {
            call.resolve(["token": NSNull()])
        }
    }

    @objc func removeToken(_ call: CAPPluginCall) {
        AppGroupStorage.shared.removeToken()
        call.resolve()
    }
}
